### ISSUE TITLE
New workspace containers can't be created once an old workspace crashes

### ISSUE THREAD

**Issue by jonatanschroeder** [PrairieLearn/PrairieLearn#3079]
Steps to reproduce:

- have a question with workspaces, that causes the workspace to crash. For example, add an invalid argument (e.g., `--invalid-arg`) to the `args` field in `info.json` for the demo for vscode.
- pull/sync question, then open its preview.
- Open Workspace. This will cause the workspace container to be created, but it will crash. The page will be stuck at "Starting container" instead of identifying that the container is no longer running (bug one).
- Close the Workspace. Fix the question `info.json`, then pull/sync question again and refresh the preview page.
- Open Workspace again. This should cause a new container to be created, now not crashing. Instead, the page includes a message `LAUNCHING - Deploying more computational resources (XX seconds elapsed)`. No container is created (bug two), though the old container still exists, though not running.

In summary, there are two issues: 

1. when the workspace container crashes, there is no way to identify it crashed other then checking the docker dashboard/command-line, and the crash doesn't remove the container, only stops it.
2. after the container crashes, no more workspace containers can be created. The only workaround I found to get new containers to be created after this is to restart the prairielearn container.

**Comment by jonatanschroeder** [comment]
I haven't seen this issue in over 2 years, so I'll close this issue.
